crates/db_schema/src/impls/community.rs

```diff
@@ -150,30 +151,16 @@ impl Community {
     for p in &posts {
       debug_assert!(p.community_id == community_id);
     }
-    conn
-      .build_transaction()
-      .run(|conn| {
-        Box::pin(async move {
-          update(
-            // first remove all existing featured posts
-            post::table,
-          )
-          .filter(post::dsl::community_id.eq(community_id))
-          .set(post::dsl::featured_community.eq(false))
-          .execute(conn)
-          .await?;
-
-          // then mark the given posts as featured
-          let post_ids: Vec<_> = posts.iter().map(|p| p.id).collect();
-          update(post::table)
-            .filter(post::dsl::id.eq_any(post_ids))
-            .set(post::dsl::featured_community.eq(true))
-            .execute(conn)
-            .await?;
-          Ok(())
-        }) as _
-      })
-      .await
+    // Mark the given posts as featured and all other posts as not featured.
+    let post_ids = posts.iter().map(|p| p.id);
+    update(post::table)
+      .filter(post::dsl::community_id.eq(community_id))
+      // This filter is just for performance
+      .filter(post::dsl::featured_community.or(post::dsl::id.eq_any(post_ids.clone())))
```

Comment:
Is there a way to avoid cloning `post_ids`?

Reply:
I don't think so. Also, `post_ids` is an iterator over a borrowed slice, so there aren't any allocations other than the ones done by `eq_any`. I think the only way to avoid calling `eq_any` twice is to remove the filter, which would cause the database to read (and probably even lock) all posts in the community.

Reply:
That's right, it's an iterator being cloned instead of a Vec. I'm not sure how cloning works for iterators tbh, but the elements are only ints, so I doubt a clone would be that expensive regardless.